src/npcmove.cpp

```diff
@@ -3401,17 +3401,17 @@ bool npc::wield_better_weapon()
     // Needs to check reload speed, RELOAD_ONE etc.
     // Until then, the NPCs should reload the guns as a last resort
 
-    if( best == weapon ) {
+    if( &best == &weapon ) {
```

Comment:
As written, this will never evaluate to true since "best" has its own separate address on the stack. But if you change "best" to a pointer then `best == &weapon` would work.